### PR TITLE
VMWare: Raw device mapping

### PR DESCRIPTION
**Pull request by sandrich**
##### SUMMARY
Raw device mapping which should tackle #24837
Had to recreate a PR to replace #62394 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
vmware_guest_disk

##### ADDITIONAL INFORMATION
The enhancement allows to map raw devices into the VM. The current implementation is working but is missing some tests and code cleanup. Also currently only physicalMode compatibility mode is supported.

Also thanks to @radudd for his contributions

```paste below

```

### CODE REVIEW

**Comment by sandrich** [comment]
@Akasurde my git skills failed, recreated the PR with your suggested change

**Comment by Akasurde** [comment]
@Tomorrow9 Could you please review this? Thanks in advance.

**Comment by sandrich** [comment]
@Tomorrow9, @Akasurde seems that the integration test fail but it has nothing to do with my implementation. I suspect that when the test runs it already has the disk presented from a previous run? Or are they cleaned after the tests?

`name: create new disk(s) with valid disk mode`

Also it would be good if we could add some integration tests for raw devices. Not sure if we have the capabilities to do that.

File: lib/ansible/modules/cloud/vmware/vmware_guest_disk.py [hunk 1]
```diff
@@ -631,9 +681,25 @@ def sanitize_disk_inputs(self):
...
+                                          "'compatibility_mode' value from ['physicalMode']." % disk_index)
+
+                    current_disk['disk_compatibility_mode'] = compatibility_mode
+
+                if 'raw_device' not in disk:
+                    self.module.fail_json(msg="raw_device needs to be defined when using disk type 'raw' for disk index [%s]" % disk_index)
+                else:
+                    current_disk['disk_raw_device'] = disk.get('raw_device')
+
+                disks_data.append(current_disk)
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_disk.py]: I tested change with my setup and looks like this line should be outside this `if` statement. This change is causing the failure in the CI test case. Could you please check this again?

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_guest_disk.py (hunk 1 above)
Comment (sandrich) [reply lib/ansible/modules/cloud/vmware/vmware_guest_disk.py]: Should be resolved. Tested rdm locally.

**Comment by goneri** [comment]
Hi!

Thank you for your PR. Since March 2, 2020, the Ansible team does not accept any new changes on the VMware modules on the core `ansible/ansible` repository.
The development continue but in the new VMware collection repository: https://github.com/ansible-collections/vmware
Could you please reopen the PR there?

Links:
- https://groups.google.com/forum/#!msg/ansible-devel/6UAfogEDtG8/wMTgxBGWCgAJ
- https://github.com/ansible-collections/overview/blob/master/README.rst

**Pull request closed without merging**